src/test/java/com/microsoft/graph/authentication/TokenCredentialAuthProviderTest.java

```diff
@@ -0,0 +1,56 @@
+package com.microsoft.graph.authentication;
+
+import com.azure.core.credential.TokenCredential;
+import com.microsoft.graph.mocks.MockTokenCredential;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.mock;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.concurrent.ExecutionException;
+
+public class TokenCredentialAuthProviderTest {
+
+    public final String testToken = "CredentialTestToken";
+
+    @Test
+    public void providerAddsTokenOnValidHostName() throws MalformedURLException, InterruptedException,
+            ExecutionException {
+        // Arrange
+        final URL url = new URL("https://graph.microsoft.com");
```

Comment:
How about a test for all of the national and public cloud endpoints (such as in the test above)?

Reply:
would it add any value though? the code it depends on is already tested for the multiple national clouds.

Reply:
In that case, we don't need this test at all, right? Because that other test also covers https://graph.microsoft.com (unless I misunderstood)?